### PR TITLE
Set up `googlesitekit-components`

### PR DESCRIPTION
**Pull request by nfmohit**
## Summary

Addresses issue:

- #5958 

## Relevant technical choices

This PR creates a new `googlesitekit-components` package for Material 2 components.

## PR Author Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

File: assets/js/googlesitekit-components-gm2.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+	Button,
+	Checkbox,
+	Chip,
+	Dialog,
+	Menu,
+	ProgressBar,
+	Radio,
+	Switch,
+	Tooltip,
+} = Components;
```
Comment (nfmohit) [inline assets/js/googlesitekit-components-gm2.js:40]: Exported each component separately for named imports to work in tests and stories.

File: tests/js/jest.config.js [hunk 2]
```diff
@@ -37,7 +37,12 @@ module.exports = {
 	// Matches aliases in webpack.config.js.
 	moduleNameMapper: {
 		// New (JSR) modules.
-		'^googlesitekit-(.+)$': '<rootDir>assets/js/googlesitekit-$1',
```
Comment (nfmohit) [inline tests/js/jest.config.js:40]: Mapped each module separately now that we have modules with varying file names (e.g. `-gm2.js`).

File: assets/js/googlesitekit-components-gm2.js (hunk 1 above)
Comment (tofumatt) [reply assets/js/googlesitekit-components-gm2.js:40]: Thanks for this, I've been meaning to do it for ages with `googlesitekit-data` 😅

**Review by tofumatt — APPROVED** [review]
Amazing! Tested locally and all seems to work, looks good to me. 👍🏻

File: tests/js/jest.config.js [hunk 3]
```diff
@@ -37,7 +37,12 @@ module.exports = {
...
+		'googlesitekit-api': '<rootDir>assets/js/googlesitekit-api',
+		'googlesitekit-data': '<rootDir>assets/js/googlesitekit-data',
+		'googlesitekit-modules': '<rootDir>assets/js/googlesitekit-modules',
+		'googlesitekit-widgets': '<rootDir>assets/js/googlesitekit-widgets',
+		'googlesitekit-components':
```
[the remark this answers is not in the document]
Comment (tofumatt) [reply tests/js/jest.config.js:51]: Add comment to filename override.

**Review by tofumatt — CHANGES_REQUESTED** [review]
Oops, so this looks great but looks like there are some merge conflicts.

Could you resolve them? After that we can get this merged 👍🏻

**Comment by nfmohit** [comment]
Thank you for the review and for adding the comments, @tofumatt! I have resolved the merge conflicts and I think this is ready to go.

**Review by tofumatt — APPROVED** [review]

**Pull request merged**